src/physics/initial_conditions/initial_condition_function.cpp

```diff
@@ -450,6 +450,241 @@ ::value(const dealii::Point<dim,real> &point, const unsigned int istate) const
     return soln_conservative[istate];
 }
 
+// ========================================================
+// Initial Condition - Euler Base
+// ========================================================
+template <int dim, int nstate, typename real>
+InitialConditionFunction_EulerBase<dim, nstate, real>
+::InitialConditionFunction_EulerBase(
+    Parameters::AllParameters const* const param)
+    : InitialConditionFunction<dim, nstate, real>()
+{
+    // Euler object; create using dynamic_pointer_cast and the create_Physics factory
+    // This test should only be used for Euler
+    this->euler_physics = std::dynamic_pointer_cast<Physics::Euler<dim, dim + 2, double>>(
+        Physics::PhysicsFactory<dim, dim + 2, double>::create_Physics(param));
```

Comment:
i just realized that this and `convert_primitive_to_conversative_value` definition is very similar to `InitialConditionFunction_TaylorGreenVortex`; could you extract these and make `InitialConditionFunction_TaylorGreenVortex` derived from `InitialConditionFunction_EulerBase`?

Reply:
Refactored the TGV initial condition to be derived from `InitialConditionFunction_EulerBase`. Ran `MPI_3D_EULER_SPLIT_TAYLOR_GREEN` and `MPI_3D_EULER_SPLIT_TAYLOR_GREEN_CURV` to verify implementation and both tests passed.

Reply:
can you please run all the tests defined in `tests/integration_tests_control_files/taylor_green_vortex_integration/CMakeLists.txt` ?

Reply:
This should be adapted to work for NS cases as well; keep the logic from the original implementation

Reply:
See the code suggestion i left in another comment

Reply:
Ran all 13 tests defined in the specified folder. They all passed! 😄 
![image](https://github.com/dougshidong/PHiLiP/assets/64045004/6db176e7-2d5b-4e8f-b169-0aae35d5e751)

Reply:
Thank you!